### PR TITLE
Document Skills API

### LINKED ISSUE #424: Document Skills API
We should document this endpoint using a combination of:
- [API Blueprint](https://apiblueprint.org/) for writing the documentation of resources, actions, and URIs
- [Dredd](https://dredd.readthedocs.io/en/latest/) for testing the documentation matches the API
- [Apiary](https://apiary.io/) for browsing the docs

Please reference the journey issue #387 for further instructions or commentary on documenting.

### PR DESCRIPTION
**Pull request by npendery**
This update adds API documentation for Skills. The following actions have been documented:
- Create a skill – `POST /skills`
- Find a skill – `GET /skills/{id}`
- List/search skills - `GET /skills`

Closes #424

### CODE REVIEW

**Comment by joshsmith** [comment]
@npendery made some changes here. Some of that affects categories and not just skills.

A couple things I think are missing here:
- filtering in the endpoint
- searching for skills (the autocomplete part)

Both are buried in the `index` action.

We can tackle those separately if we add some issues for them.

**Comment by npendery** [comment]
@JoshSmith just wondering why you decided to use this PR to change category items? Wouldn't we want to separate that out in a separate PR?

**Comment by joshsmith** [comment]
@npendery was just for simplicity's sake. I normally would want to do that but was having trouble organizing thoughts around your PR without the organizational work there.

File: blueprint/api.apib [hunk 1]
```diff
@@ -595,6 +662,26 @@ This allows A Skill to be under different Roles and a Role to have multiple Skil
...
++ data(Skill Resource)
+
+## Skills Response (object)
++ data(array[Skill Resource])
+
+## Skill Resource (object)
++ include Skill Resource Identifier
++ attributes
+  + title: `Ruby`
+  + description: `Knowledgeable about the Ruby Programming Language`
```
Comment (JoshSmith) [inline blueprint/api.apib]: Why'd you change this from the Elixir implementation I had pushed up? Thought the description there was clearer (and would be more fun for our new Elixir folks).

**Comment by joshsmith** [comment]
Did we lose a number of changes here? I feel like there's a lot missing in what I had made and can't tell since it was rebased.

**Comment by joshsmith** [comment]
Yeah, this lost all the very real changes I was trying to make to suggest some rework of grouping of the resources. Not sure how to proceed.

**Comment by npendery** [comment]
Oh crap. Not sure how that happened....

**Comment by joshsmith** [comment]
@npendery is this good to review again? If it was, then I approve.

**Comment by joshsmith** [comment]
I think the `query` and the `filter` should be broken out into two separate use cases, or (and maybe better) we should spend some time defining what each does in plain English a little better.

I can tackle that if you'd like, just assign to me.

**Comment by npendery** [comment]
@JoshSmith yeah, wasn't sure how we would want to display that. Will update and separate them out.

**Comment by rigelstpierre** [comment]
🐑 🇮🇹
